contracts/vault/packages/VaultPackage.sol

```diff
@@ -1391,27 +1402,16 @@ contract VaultPackage is VaultStorage, IVault, IVaultInit, IVaultEvents {
         // Remove strategy if it is in the default queue.
         uint256 defaultQueueLength = defaultQueue.length;
         if (defaultQueueLength > 0) {
-            uint256 newDefaultQueueLength = defaultQueueLength;
-            // Find if there is a need to reduce the queue length
             for (uint256 i = 0; i < defaultQueueLength; i++) {
                 if (defaultQueue[i] == strategy) {
-                    newDefaultQueueLength--;
-                    break;
-                }
-            }
-            // If the queue length needs to be reduced, create a new queue with the new length
-            if (newDefaultQueueLength != defaultQueueLength) {
-                address[] memory newQueue = new address[](newDefaultQueueLength);
-                uint256 k;
-                for (uint256 i = 0; i < defaultQueueLength; i++) {
-                    address _strategy = defaultQueue[i];
-                    // Add all strategies to the new queue besides the one revoked.
-                    if (_strategy != strategy) {
-                        newQueue[k++] = _strategy;
+                    // Shift all elements down one position from the point of removal
+                    for (uint256 j = i; j < defaultQueueLength - 1; j++) {
```

Comment:
Now I am thinking that it could be better to just change the removed element to the last one, but lets keep the current approach. It may make sense if queue is expected to be in certain order and we are willing to keep it unchanged